go-controller/pkg/kubevirt/dhcp.go

```diff
@@ -22,17 +22,71 @@ const (
 	dhcpLeaseTime = 3500
 )
 
+type DHCPConfigsOpt = func(*dhcpConfigs)
+
 type dhcpConfigs struct {
 	V4 *nbdb.DHCPOptions
 	V6 *nbdb.DHCPOptions
 }
 
+func WithIPv4Router(router string) func(*dhcpConfigs) {
+	return func(configs *dhcpConfigs) {
+		if configs.V4 == nil {
+			return
+		}
+		configs.V4.Options["router"] = router
+	}
+}
+
+func WithIPv4MTU(mtu int) func(*dhcpConfigs) {
+	return func(configs *dhcpConfigs) {
+		if configs.V4 == nil {
+			return
+		}
+		configs.V4.Options["mtu"] = fmt.Sprintf("%d", mtu)
+	}
+}
+
+func WithIPv4DNServer(dnsServer string) func(*dhcpConfigs) {
+	return func(configs *dhcpConfigs) {
+		if configs.V4 == nil {
+			return
+		}
+		configs.V4.Options["dns_server"] = dnsServer
+	}
+}
+
+func WithIPv6DNServer(dnsServer string) func(*dhcpConfigs) {
+	return func(configs *dhcpConfigs) {
+		if dnsServer == "" {
```

Comment:
is very common to have dual stack but IPv4 DNS server since the DNS protocol is able to answer names for IPv6 over ipv4, A record versus AAA record.

don't know if we want to be prepare for single stack IPV6 though.